### DISCUSSION TITLE
findMany users of multiple unknown property key values

### DISCUSSION THREAD

**Discussion by LewisMorgans** [prisma/prisma#10585] [Q&A] [answered]
Hey, I'm new to the prisma world. 

I'm trying to use a for loop with key values of an argument to search the database of users, for an instance of a user that matches all of the object property values sent in the argument. 

I'm clinging at straws to be honest at the moment. 

so for example, inside my collection.

```
user {
name: jim
python: true,
typescript: true
}

user {
name: bob
python: false,
typescript: true
}
```

And I want to search the collection, for a user that has both python and typescript with true values. I can't statically type these as there are many of them in actuality and the argument provided is fluid. So my attempt so far was below. But it doesn't work, I imagine it's because it does find a user with one of the arguments and just breaks the loop.

The argument passed to the function is an object as demonstrated below:

```
filters = {
python: true,
typescript: true
}
```

(this.getUsers is just a findMany function returning all users of  a type of role)

```
  public async filterUsers(filters) {
    return this.getUsers('developer')
      .then(async (users) => {
        for (const [key, value] of Object.entries(filters)) {
          return await prisma.user.findMany({
            where: {
              [key]: {
                equals: !!value,
              },
            },
          });
        }
      })
      .catch((error) => error);
  }
```

I'm not sure what to do and I can't find any documentation supporting this use case. If anyone could lend a hand, that would be superb. Happy to explain more if needed. Cheers.

**Comment by austincrim** [comment]
Hey @LewisMorgans 👋!

Would you mind sharing the relevant parts of your `schema.prisma` file so I can dig into this a bit more?

**Reply by LewisMorgans** [reply]
Hi Austincrim,

Thanks for taking the time. I managed to figure it out, I couldn't get it working without having another model called options to use in a 1:1 relation with the user model. I've popped my answer down below if you'd like to see it 👍

**Comment by LewisMorgans** [comment, marked as answer]
In case it helps anyone else here is the answer when using a 1:1 relation.

### **Method**
```
  public async filterUsers(options: object): Promise<object> {
    for (const [key, value] of Object.entries(options)) {
      return await prisma.user.findMany({
        where: {
          options: {
            [key]: value,
          },
        },
      });
    }
  }
```

### **SCHEMA**
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = [ "mongoDb", "filterJson"]
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
    id       String @id @default(dbgenerated()) @map("_id") @db.ObjectId
    email    String
    password String
    role     String
    options Options?
}

model Options {
    user   User   @relation(fields: [userId], references: [id])
    userId String @unique @id @map("_id") @db.ObjectId
    typeScript Boolean
    rxjs Boolean
 }
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
